File: NIS11/app/src/main/java/com/example/nis11/MainActivity.java
package com.example.nis11;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.app.AlertDialog;
import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.ContentResolver;
import android.content.Context;
import android.content.DialogInterface;
import android.content.pm.PackageManager;
import android.media.AudioAttributes;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button button = findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Context context = MainActivity.this;
                NotificationChannel newnotchan = null;
                newnotchan = new NotificationChannel("mychannel1", "mychannel", NotificationManager.IMPORTANCE_HIGH);
                AudioAttributes audioAttributes = new AudioAttributes.Builder().setContentType(AudioAttributes.CONTENT_TYPE_MUSIC).setUsage(AudioAttributes.USAGE_MEDIA).build();
                newnotchan.setSound(Uri.parse(ContentResolver.SCHEME_ANDROID_RESOURCE + "://" + getPackageName() + "/" + R.raw.sound), audioAttributes);
                NotificationManager notificationManager = (NotificationManager) getApplicationContext().getSystemService(Context.NOTIFICATION_SERVICE);
                notificationManager.createNotificationChannel(newnotchan);
                Notification notification = null;
                notification = new Notification.Builder(context, "mychannel1")
                        .setContentTitle("Уведомление от уведомителя").setContentText("Друг, ты уведомлен")
                        .setTicker("new notification!").setChannelId("mychannel1").
                                setSmallIcon(android.R.drawable.ic_dialog_alert).setOngoing(true).build();

                notificationManager.notify(0, notification);
            }
        });


    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        AlertDialog.Builder dialog = new AlertDialog.Builder(MainActivity.this);
        try {
            dialog.setMessage(getTitle().toString() + " версия " + getPackageManager().getPackageInfo(getPackageName(),
                    0).versionName + "\r\n\nПрограмма с примером выполнения диалогового окна \r\n\nАвтор -  Шагаров Дмитрий Александрович БПИ202");
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }
        dialog.setTitle("О программе");
        dialog.setNeutralButton("OK", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                dialog.dismiss();
            }
        });
        dialog.setIcon(R.mipmap.ic_launcher_round);
        AlertDialog alertDialog = dialog.create();
        alertDialog.show();
        return super.onOptionsItemSelected(item);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu, menu);
        return super.onCreateOptionsMenu(menu);
    }
}

File: NIS1-2/app/src/main/java/com/example/nis/MainActivity.kt
package com.example.nis

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val userTextInput: EditText = findViewById(R.id.inputText)
        val userTextOutput: TextView = findViewById(R.id.userText)
        val userTextButton: Button = findViewById(R.id.inputButton)
        userTextButton.setOnClickListener(View.OnClickListener() {
            userTextOutput.text = if (userTextInput.text.toString() != "") {
                userTextInput.text.toString()
            } else {
                "Шляпа пуста :("
            }
        })

//        input = findViewById(R.id.inputText)
//        output = findViewById(R.id.userText)
    }

//    fun sendText(view: View) {
//        output.text = if (input.text.toString() != "") {
//            input.text.toString()
//        } else {
//            "Шляпа пуста :("
//        }
//    }
}